frontend/src/composables/useShootContext/index.js

```diff
@@ -0,0 +1,1261 @@
+  const kubernetesVersion = computed({
+    get () {
+      return get(manifest.value, 'spec.kubernetes.version')
+    },
+    set (value) {
+      set(manifest.value, 'spec.kubernetes.version', value)
+    },
+  })
+
+  function resetKubernetesVersion () {
+    const defaultKubernetesVersionDescriptor = cloudProfileStore.defaultKubernetesVersionForCloudProfileName(cloudProfileName.value)
+    kubernetesVersion.value = get(defaultKubernetesVersionDescriptor, 'version')
+  }
+
+  const kubernetesEnableStaticTokenKubeconfig = computed({
+    get () {
+      return !!get(manifest.value, 'spec.kubernetes.enableStaticTokenKubeconfig')
+    },
+    set (value) {
+      set(manifest.value, 'spec.kubernetes.enableStaticTokenKubeconfig', !!value)
+    },
+  })
+
+  function resetKubernetesEnableStaticTokenKubeconfig () {
+    kubernetesEnableStaticTokenKubeconfig.value = false
+  }
+
+  function resetKubernetes () {
+    resetKubernetesVersion()
+    resetKubernetesEnableStaticTokenKubeconfig()
+  }
+
+  /* cloudProfileName */
+  const cloudProfileName = computed({
+    get () {
+      return get(manifest.value, 'spec.cloudProfileName')
+    },
+    set (value) {
+      set(manifest.value, 'spec.cloudProfileName', value)
+      resetNetworkingType()
+      resetKubernetesVersion()
+      resetSecretBindingName()
+      resetRegion()
+      resetProviderControlPlaneConfigLoadBalancerClasses()
+      resetProviderInfrastructureConfigProjectID()
+      resetProviderInfrastructureConfigFirewallImage()
+    },
+  })
+
+  function resetCloudProfileName () {
+    const cloudProfile = head(cloudProfiles.value)
+    cloudProfileName.value = get(cloudProfile, 'metadata.name')
+  }
+
+  /* secretBindingName */
+  const secretBindingName = computed({
+    get () {
+      return get(manifest.value, 'spec.secretBindingName')
+    },
+    set (value) {
+      set(manifest.value, 'spec.secretBindingName', value)
+      resetPurpose()
+    },
+  })
+
+  const infrastructureSecret = computed({
+    get () {
+      return find(infrastructureSecrets.value, ['metadata.name', secretBindingName.value])
+    },
+    set (value) {
+      secretBindingName.value = get(value, 'metadata.name')
+    },
+  })
+
+  function resetSecretBindingName () {
+    infrastructureSecret.value = head(infrastructureSecrets.value)
+  }
+
+  /* networking */
+  const networkingNodes = computed(() => {
+    return get(manifest.value, 'spec.networking.nodes')
+  })
+
+  const networkingType = computed({
+    get () {
+      return get(manifest.value, 'spec.networking.type')
+    },
+    set (value) {
+      set(manifest.value, 'spec.networking.type', value)
+    },
+  })
+
+  function resetNetworkingType () {
+    networkingType.value = head(gardenerExtensionStore.networkingTypes)
+  }
+
+  /* region */
+  const region = computed({
+    get () {
+      return get(manifest.value, 'spec.region')
+    },
+    set (value) {
+      set(manifest.value, 'spec.region', value)
+      resetProviderWorkers()
```

Comment:
```suggestion
      if (!workerless.value) {
        resetProviderWorkers()
      }
```
We should not reset the workers in case workerless is enabled, as this would make the cluster have workers again.
However, if we don't do it here we need to ensure it is done somewhere further up in the call hierarchy, eg. by adding `resetProviderWorkers()` to the `cloudProfileName` computed property setter.
Not sure if this is the right way to do it though

Reply:
I think we need to have an explicit property for the `workerless` state and not derive it from the existing numberOfWorkers. I think I understood the problem.